File: C/BestTimeToBuyAndSellStock-leetcode121.c
int maxProfit(int *prices, int pricesSize)
{
    int l = 0, r = 1;
    int maxP = 0;
    
    while (r < pricesSize) {
        if (prices[r] > prices[l])
            maxP = prices[r] - prices[l] > maxP ? prices[r] - prices[l] : maxP;
        else
            l = r;
        r++;
    }
    return maxP;
}

File: Python/BestTimeToBuyAndSellStock-leetcode121.py
class Solution:
    def maxProfit(self, prices: list[int]) -> int:
        l, r = 0, 1
        maxP = 0

        while r < len(prices):
            if prices[l] < prices[r]:
                profit = prices[r] - prices[l]
                max = (profit if profit > max else max)
            else:
                l = r
            r += 1
        return max
